src/Exception/InvalidIriException.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Exception;
+
+class InvalidIriException extends InvalidArgumentException
+{
+    private $reason;
+
+    public function __construct(int $code = 0, \Throwable $previous = null)
+    {
+        parent::__construct("Invalid value provided (Invalid IRI?).", $code, $previous);
```

Comment:
Can't we just remove everything in this class and call the exception with a message?